docs_src/design/adr/device-service/0011-DeviceService-Rest-API.md

```diff
@@ -0,0 +1,174 @@
+| Boolean | `bool` | "true" or "false"
+| Integer | `uint8-uint64`, `int8-int64` | Numeric string, eg "-132"
+| Float | `float32`, `float64` | base64 encoded little-endian binary or decimal with exponent, eg "1.234e-5"
+| String | `string` | string
+| Binary | `bytes` | octet array
+| Array | `boolarray`, `uint8array-uint64array`, `int8array-int64array`, `float32array`, `float64array` | JSON Array, eg "["1", "34", "-5"]"
+
+Notes:
+- The presence of a Binary reading will cause the entire Event to be encoded using CBOR rather than JSON
+- The representation of a float is determined by the *"floatEncoding"* attribute of the device resource. *"base64"* is the default, *"eNotation"* will cause the float to be written as a decimal with exponent.
+
+#### Readings and Events
+
+A Reading represents a value obtained from a deviceResource. It contains the following fields
+
+| Field name | Description
+| --- | ---
+| *name* | The name of the deviceResource
+| *valueType* | The type of the data
+| *origin* | A timestamp indicating when the reading was taken
+| *value* | The reading value
+| *mediaType* | (Only for Binary readings) The MIME type of the data
+| *floatEncoding* | (Only for floats and arrays of floats) The float representation in use
+
+An Event represents the result of a `GET` command. If the command names a deviceResource, the Event will contain a single Reading. If the command names a deviceCommand, the Event will contain as many Readings as there are deviceResources listed in the deviceCommand.
+
+The fields of an Event are as follows:
+
+| Field name | Description
+| --- | ---
+| *device* | The name of the Device from which the Readings are taken
+| *origin* | The time at which the Event was created
+| *readings* | An array of Readings
+
+#### Query Parameters
+
+Calls to the device endpoints may include a Query String in the URL. This may be used to pass parameters relating to the request to the device service. Individual device services may define their own parameters to control specific behaviors. Parameters beginning with the prefix `ds-` are reserved to the Device SDKs and the following parameters are defined for GET requests:
+
+| Parameter | Valid Values | Default | Meaning
+| --- | --- | --- | ---
+| *ds-postevent* | "yes" or "no" | "no" | If set to yes, a successful `GET` will result in an event being posted to core-data
+| *ds-returnevent* | "yes" or "no" | "yes" | If set to no, there will be no Event returned in the http response
+
+#### Device States
+
+A Device in EdgeX has two states associated with it: the Administrative state and the Operational state. The Administrative state may be set to `LOCKED` (normally `UNLOCKED`) to block access to the device for administrative reasons. The Operational state may be set to `DISABLED` (normally `ENABLED`) to indicate that the device is not currently working. In either case access to the device via this endpoint will be denied and HTTP 403 ("Forbidden") will be returned.
```

Comment:
You can set it in core-metadata, eg `curl -X PUT -d '{"operatingstate":"disabled"}' 0:48081/api/v1/device/67fba75c-12fa-4978-be58-dc3cd71a9ffc`

Reply:
In the above table, this situation should return 423 Locked instead of 403 Forbidden.


423 | the specified device is locked (admin state) or disabled (operating state)
-- | --

Reply:
Yes. Corrected.